src/mono/mono/metadata/marshal.c

```diff
@@ -4702,7 +4702,7 @@ get_virtual_stelemref_kind (MonoClass *element_class)
 	if (element_class == mono_defaults.object_class)
 		return STELEMREF_OBJECT;
 	if (is_monomorphic_array (element_class))
-		return STELEMREF_SEALED_CLASS;
+		return STELEMREF_COMPLEX;
```

Comment:
This is going to make all array-of-array checks go to the slow isinst check, isn't it?

Can we make `is_monomorphic_array` more sensitive instead: maybe if its element is a function pointer, then return false?

Reply:
Fixed.